tests/platform_tests/test_process_reboot_cause.py

```diff
@@ -0,0 +1,190 @@
+
+        return reboot_cause_file
+
+    def remove_json_file(self, reboot_cause_file):
+        logging.info("Clean up the newly created JSON file: {}".format(reboot_cause_file))  # noqa: E501
+        _ = self.duthost.command('rm -f {}'.format(reboot_cause_file))
+
+    def bad_json_file_assertions(self, status, result, journalctl_output):
+
+        # Check for exeception if image version < 2023110.20 , 20230531.33 or 20220531.53  # noqa: E501
+        if self.image_major_ver < 20220531 or \
+          (self.image_major_ver == 20231110 and self.image_minor_ver < 20) or \
+          (self.image_major_ver == 20230531 and self.image_minor_ver < 33) or \
+          (self.image_major_ver == 20220531 and self.image_minor_ver < 53):
+
+            logging.info("OS Version {} does not have the patched process-reboot-cause file with try-except block".format(self.image_ver))  # noqa: E501
+            is_json_decode_error = any(JSON_DECODE_ERROR in line for line in journalctl_output)  # noqa: E501
+
+            pytest_assert(result == "exit-code")
+            pytest_assert(status == 1)
+            pytest_assert(is_json_decode_error)
+
+        else:
+            logging.info("OS Version {} has the patched process-reboot-cause file with try-except block".format(self.image_ver))  # noqa: E501
+            exception_handled_log = any(EXCEPTION_HANDLED_SYSLOG in line for line in journalctl_output)  # noqa: E501
+            logging.info("exception_handled_log: {}".format(journalctl_output))
+            pytest_assert(result == "success")
+            pytest_assert(status == 0)
+            pytest_assert(exception_handled_log)
+
+    def good_json_file_assertions(self, status, result, cause, journalctl_output, statedb_output):  # noqa: E501
+
+        excpected_reboot_cause = any(cause in line for line in statedb_output)  # noqa: E501
+
+        pytest_assert(result == "success")
+        pytest_assert(status == 0)
+        pytest_assert(excpected_reboot_cause)
+
+    def process_reboot_cause_file_runner(self, filetype=""):
+
+        # Create a JSON file
+        reboot_cause_file = self.create_json_file(filetype)
+
+        # Restart process-reboot-cause.service
+        self.restart_service()
+
+        # Wait 5 seconds
+        time.sleep(5)
+
+        # Get latest status of the service
+        logging.info("Get latest status of the process-reboot-cause.service")
+        main_pid, status, result = self.get_service_status()
+        journalctl_output = self.duthost.command(JOURNALCTL_CMD.format(main_pid))["stdout_lines"]  # noqa: E501
+
+        # Get STATE_DB kvps created from latest JSON file
+        statedb_output = self.duthost.command('redis-cli -n 6 HGETALL "REBOOT_CAUSE|9999_99_99_99_99_99"')["stdout_lines"]  # noqa: E501
+        _ = self.duthost.command('redis-cli -n 6 DEL "REBOOT_CAUSE|9999_99_99_99_99_99"')  # noqa: E501
+
+        # Remove the JSON file
```

Comment:
@assrinivasan How are we making sure we are backing up the reboot cause history file before the test start and restoring it after the test is completed?

Reply:
We do not need to back up all the existing reboot cause files because the actual reboot cause files are untouched. In this test, we are creating a dummy file (good or malformed), running a test on it, verifying its effects on the process-reboot-cause service + the STATE_DB and then immediately deleting the newly created reboot cause file.